landscape/client/snap_utils.py

```diff
@@ -4,6 +4,37 @@
 from landscape.client.snap_http import SnapdHttpException
 
 
+class IgnoreYamlAliasesLoader(yaml.SafeLoader):
+    """Patch `yaml.SafeLoader` to ignore aliases like *alias when loading.
+
+    For instance, a system-user assertion can have the following json:
+      {
+          [...]
+           "system-user-authority": "*",
+          [...]
+      }
+    which after signing gets converted to yaml that looks like:
+      [...]
+      system-user-authority: *
+      [...]
+    pyyaml tries to parse the * as the start of an alias leading to errors.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.yaml_implicit_resolvers.pop("*", None)
```

Comment:
I hate to continue to pick on this, but `yaml_implicit_resolvers` seems to be a class attribute on `SafeLoader`, so `pop`ping on it is mutative. See:

```python
>>> import yaml
>>> yaml.SafeLoader.yaml_implicit_resolvers["*"]
[('tag:yaml.org,2002:yaml', re.compile('^(?:!|&|\\*)$'))]
>>> class MyLoader(yaml.SafeLoader):
...     def __init__(self, *args, **kwargs):
...         super().__init__(*args, **kwargs)
...         self.yaml_implicit_resolvers.pop("*", None)
... 
>>> import io
>>> my_loader = MyLoader(io.StringIO())
>>> yaml.SafeLoader.yaml_implicit_resolvers["*"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: '*'
```

I think the proper way to exempt this resolver's behaviour is either to override `resolve` (too complex to do here, probably) or override with a copy and mutate that copy.

Reply:
Copying seems to work:

```python
In [1]: import io
   ...: import yaml
   ...: from copy import deepcopy
   ...:

In [2]: yaml.SafeLoader.yaml_implicit_resolvers["*"]
Out[2]: [('tag:yaml.org,2002:yaml', re.compile(r'^(?:!|&|\*)$', re.UNICODE))]

In [3]: class MyLoader(yaml.SafeLoader):
   ...:     def __init__(self, *args, **kwargs):
   ...:         super().__init__(*args, **kwargs)
   ...:         self.yaml_implicit_resolvers = deepcopy(super().yaml_implicit_resolvers)
   ...:         self.yaml_implicit_resolvers.pop("*", None)
   ...:

In [4]: my_loader = MyLoader(io.StringIO())

In [5]: yaml.SafeLoader.yaml_implicit_resolvers["*"]
Out[5]: [('tag:yaml.org,2002:yaml', re.compile(r'^(?:!|&|\*)$', re.UNICODE))]
```

Lemme switch to that.